src/main/java/marquez/db/DatasetDao.java

```diff
@@ -40,40 +40,45 @@
   @CreateSqlObject
   DbTableVersionDao createDbTableVersionDao();
 
-  @SqlUpdate(
+  @SqlQuery(
       "INSERT INTO datasets (guid, namespace_guid, datasource_uuid, urn, description) "
-          + "VALUES (:uuid, :namespaceUuid, :datasourceUuid, :urn, :description)")
-  void insert(@BindBean DatasetRow datasetRow);
+          + "VALUES (:uuid, :namespaceUuid, :datasourceUuid, :urn, :description) "
+          + "RETURNING *")
+  Optional<DatasetRow> insertAndGet(@BindBean DatasetRow datasetRow);
 
+  @Deprecated
   @Transaction
   default void insertAll(
       DatasourceRow datasourceRow,
       DatasetRow datasetRow,
       DbTableInfoRow dbTableInfoRow,
       DbTableVersionRow dbTableVersionRow) {
     createDatasourceDao().insert(datasourceRow);
-    insert(datasetRow);
+    insertAndGet(datasetRow);
     createDbTableVersionDao().insertAll(dbTableInfoRow, dbTableVersionRow);
     updateCurrentVersion(datasetRow.getUuid(), Instant.now(), dbTableVersionRow.getUuid());
   }
 
+  @SqlQuery("SELECT EXISTS (SELECT 1 FROM datasets WHERE urn = :value)")
+  boolean exists(@BindBean DatasetUrn datasetUrn);
+
   @SqlUpdate(
-      "UPDATE datasets SET updated_at = :updatedAt, current_version_uuid = :currentVersion "
-          + "WHERE guid = :uuid")
+      "UPDATE datasets SET updated_at = :updatedAt, current_version_uuid = :currentVersionUuid "
+          + "WHERE uuid = :uuid")
   void updateCurrentVersion(UUID uuid, Instant updatedAt, UUID currentVersion);
 
-  @SqlQuery("SELECT * FROM datasets WHERE uuid = :uuid")
+  @SqlQuery("SELECT * FROM datasets WHERE guid = :uuid")
```

Comment:
Is the field on datasets `uuid` or `guid`?

Reply:
`guid`! I'll also add a test